### PR TITLE
Feat/icon ts conversion

### PR DESCRIPTION
**Pull request by EdwardIrby**
Typescript conversion, 3rd PR Attempt.

### CODE REVIEW

File: packages/icon/package.json [hunk 1]
```diff
@@ -12,23 +12,25 @@
...
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --target es5 --outDir dist/cjs",
     "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
-    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
+    "build:esm": "tsc --project tsconfig.build.json --module es2015 --target es5 --outDir dist/esm",
     "prebuild:svg": "yarn clean:svg",
     "build:svg": "node bin/icon -s src/svg -d src/react/icons core",
     "build:watch": "yarn build:js -- --watch",
     "clean:svg": "node bin/icon -s src/svg --clean",
     "copy:svg": "node cli/copy-svgs.js",
+    "prepublish": "yarn build",
```
Comment (jaketrent) [inline packages/icon/package.json:24]: I think Dane recently removed all these for the build's sake:

https://github.com/pluralsight/design-system/pull/1311/files#diff-142aada5767479fc0f7c069ae0e70ee5L18

File: packages/icon/src/css/index.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {
   colorsYellow
 } from '@pluralsight/ps-design-system-core'
 
-import * as vars from '../vars/index.js'
+import * as vars from '../vars/'
```
Comment (jaketrent) [inline packages/icon/src/css/index.ts:9]:
```suggestion
import * as vars from '../vars'
```
nit

File: packages/icon/src/react/index.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { css } from 'glamor'
```
Comment (jaketrent) [inline packages/icon/src/react/index.tsx:1]: No diff 😭

File: packages/icon/src/react/index.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+interface IconComponent
+  extends RefForwardingComponent<IconProps, HTMLDivElement, IconStatics> {}
+
+const Icon = React.forwardRef<HTMLDivElement, IconProps>(
+  ({ children, color, size, ...props }, ref) => (
+    <div
+      {...style.icon({ color, size })}
+      {...filterReactProps(props)}
```
Comment (jaketrent) [inline packages/icon/src/react/index.tsx:39]: Is there a reason to leave this one? Generally, I think these can go away.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -50,6 +50,7 @@ lint:
 test:
   stage: test
   script:
+    - find packages/icon/src/react/icons -name "*.dist.tsx" -type f  -delete
```
Comment (EdwardIrby) [inline .gitlab-ci.yml:53]: I think this should fix the problem if the first build passes and then I merge master on top. That's how #1313 failed.

**Review by jaketrent — COMMENTED** [review]
tsc and yarn build look nice!

For some reason, I can't see storybook examples.  Maybe double check that.

Screen Shot 2020-10-08 at 2 42 29 PM

File: .gitlab-ci.yml (hunk 5 above)
Comment (jaketrent) [reply .gitlab-ci.yml:53]: awesome find!

**Comment by EdwardIrby** [comment]
# Problem
CI keeps failing in the same manner and I have not been able to reproduce locally even when setting 

# Things done to make this pass
- Setting Icon version number to 18.2.3 ( Worked until recent changes on master)
- Setting Icon version number to 19.0.0 ( Worked until recent changes on master)

# Things attempted to make this work
- yarn config set workspaces-experimental true
- removing generated icons in `packages/icon/src/react/icons` before running build
- Changing static from being pulled from base Icon to directly pulling them from `packages/icon/src/vars/`
- Giving each generate Icon uniquely named interfaces
- using the same prop interface for all generated Icons
-  running `yarn workspace @pluralsight/ps-design-system-icon run build` before build
- setting strict mode to true

**Pull request closed without merging**